llvm/include/llvm/SYCLLowerIR/CallgraphUtils.h

```diff
@@ -0,0 +1,48 @@
+//===------------ CallgraphUtils.h - Callgraph utility functions
+//------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Utility functions for traversing callgraphs.
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/Function.h"
+
+#include <functional>
+namespace llvm {
+namespace CallgraphUtils {
```

Comment:
Nit: I would suggest to have some more generic names so that other SYCL-related IR processing utils could be put in the same file. (`llvm::sycl::utils`?)

Reply:
fixed thanks